src/utils/dayViewLayout/event.js

```diff
@@ -0,0 +1,184 @@
+import { accessor as get } from '../accessors'
+import dates from '../dates'
+
+export function startsBefore(date, min) {
+  return dates.lt(dates.merge(min, date), min, 'minutes')
+}
+
+export function positionFromDate(date, min, total) {
+  if (startsBefore(date, min)) {
+    return 0
+  }
+
+  const diff = dates.diff(min, dates.merge(min, date), 'minutes')
+  return Math.min(diff, total)
+}
+
+export class Event {
```

Comment:
it seems like many of the getters here are essentially `readonly` at object construction? maybe it'd be worth doing some of the calculations then so they aren't run more than once? Does `data` change? Maybe it's over optimization, but it might simplify the compiled code if nothing else

Reply:
That's definitely possible, I haven't really done any sort of optimizations yet...

Reply:
I should probably do this before merging though since it's an easy change 👍

Reply:
This was a really good call, I even ended up removing the extra MultiDayEvent class.